Add export_stats feature for JSON and CSV output

appraneethreddy opened the pull request:
This adds an export_stats method to support exporting profiling data to JSON or CSV, and includes a test script to validate the output

vigsun19 commented on base_profiler_export_test.py:
```diff
@@ -0,0 +1,36 @@
+import logging
```
why is this named '_test'?

appraneethreddy replied:
Thanks for pointing that out, you're right, the `_test` suffix was misleading. It wasn’t an actual test file but a core feature implementation. I've renamed it to `exportable_profiler.py` to better reflect its purpose. All corresponding imports have also been updated.

vigsun19 replied:
Could we rename it to something like export_stats.py to be cleaner. Also this needs to be moved into the /smartprofiler module

vigsun19 replied:
Also hitting a file not found error on export - FileNotFoundError: [Errno 2] No such file or directory: 'examples/example_output.json' could we update this to create a file if not present already?

appraneethreddy replied:
> Could we rename it to something like export_stats.py to be cleaner. Also this needs to be moved into the /smartprofiler module

Updated the filename from exportable_profiler.py to export_stats.py as suggested.
The file has also been moved to the /smartprofiler module for better alignment with its purpose.

appraneethreddy replied:
> Also hitting a file not found error on export - FileNotFoundError: [Errno 2] No such file or directory: 'examples/example_output.json' could we update this to create a file if not present already?

Handled the FileNotFoundError by updating the export_stats method to automatically create the directory if it doesn't exist using os.makedirs.

vigsun19 commented on base_profiler_export_test.py:
- Add an example usage into the /examples folder
- Also base_profiler_export could be renamed to something better

appraneethreddy replied:
Thanks for mentioning so, Both points have been addressed.

1. Have added a usage example at 'examples/export_stats_example.py' showing how to use 'ExportableProfiler' and export stats to JSON/CSV.
2. Also, renamed 'base_profiler_export_test.py' to 'exportable_profiler.py' to better represent its functionality.

Let me know if you'd like the example expanded or renamed further!

appraneethreddy committed:
Add ExportableProfiler class and test script for exporting profiling stats

appraneethreddy committed:
Renamed base_profiler_export_test.py to exportable_profiler.py and updated all imports

appraneethreddy committed:
Added example usage and export output files to examples folder

appraneethreddy committed:
Renamed and updated ExportStats with directory creation support

appraneethreddy committed:
Updated imports to use ExportStats in example and test files

appraneethreddy committed:
Delete examples/export_stats_example.py

appraneethreddy committed:
Delete examples/example_output.csv

appraneethreddy committed:
Delete examples/example_output.json

appraneethreddy committed:
Delete tests/test_export_stats.py

appraneethreddy committed:
Delete exported_stats.csv

appraneethreddy committed:
Delete exported_stats.json

appraneethreddy committed:
Updated export_stats.py

File name refrained as requested.
Updated functionality with better logging and self_stats parameter to sync with all the profilers.

appraneethreddy committed:
UnitCases file

Add test_export_stats.py via web upload
This unit test validates the functionality of the ExportStats class by ensuring profiling data from multiple profilers (CPU, memory, disk, function, and network) can be successfully exported to both JSON and CSV formats. It confirms that the exported files are created correctly and contain valid, structured data.

appraneethreddy committed:
Outputs of test cases

This file captures the console output of the unit tests, including profiler logs and test results, to verify that all profilers executed successfully and data was exported as expected.

appraneethreddy committed:
Add files via upload

This script profiles the compute_fibonacci function using CPU, memory, and function call profilers from SmartProfiler. It collects performance metrics and exports each profiler’s results separately into JSON and CSV files. The output helps analyze execution time, memory usage, and function call behavior during recursion.

appraneethreddy committed:
Add files via upload

Stats collected using  cpu_profiler for compute_fibonacci() in both CSV and Json format.

appraneethreddy committed:
Add files via upload

Commit summary
Stats collected using  func_profiler for compute_fibonacci() in both CSV and Json format.

appraneethreddy committed:
Add files via upload

Commit summary
Stats collected using  mem_profiler for compute_fibonacci() in both CSV and Json format.

appraneethreddy commented:
Build same functionality in different PR
